chain/chain/src/resharding/manager.rs

```diff
@@ -23,15 +27,27 @@ pub struct ReshardingManager {
     /// A handle that allows the main process to interrupt resharding if needed.
     /// This typically happens when the main process is interrupted.
     pub resharding_handle: ReshardingHandle,
+    /// Takes care of performing resharding on the flat storage.
+    pub flat_storage_resharder: Option<FlatStorageResharder>,
```

Comment:
I would prefer having this non-optional (maybe add a TODO as it is now convenient for this PR) in the future since as far as I know, the resharding will fail if this is not initialized yet? or can it still succeed?

Reply:
Would too prefer to have this non-optional. I think we should address this as part of this PR itself. For the places where we don't have a sender or don't need to pass a sender, we can just pass in noop() sender.

Reply:
I'll try to remove the Option and pass a noop()!